pkg/vsphere/extraconfig/decode.go

```diff
@@ -208,11 +213,21 @@ func decodeWithPrefix(kv map[string]string, dest interface{}, prefix string) int
 
 				// split v and iterate over it
 				for _, value := range strings.Split(v, "|") {
-					if field.Type().Elem().Kind() == reflect.Struct {
+					// if the map value is a pointer to a struct
```

Comment:
I think I'm confused (will look in more details) but aren't we handle this inside the outer reflect.Ptr case?